File: 3.java
import java.util.*;
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i = 0, j = 0, max = 0;
        int len = s.length();
        Set<Character> tracker = new HashSet();
        while(i <len && j < len){
            if(!tracker.contains(s.charAt(i))){
                tracker.add(s.charAt(i));
                i++;
                max = Math.max(max, tracker.size());
            }
            else{
                tracker.remove(s.charAt(j));
                j++;
            }
        }
        return max;
    }
}

File: 3.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        tracker = {}
        start, end, maxlen = 0, 0, 0
        while start < len(s) and end < len(s):
            if s[end] not in tracker:
                tracker[s[end]] = 1
                maxlen = max(maxlen, len(tracker))
                end+=1
            else:
                tracker.pop(s[start], None)
                start+=1
        return maxlen
        